src/app/data.service.ts

```diff
@@ -1,164 +1,57 @@
-
-    tableInputEng2.team = teamEng2;
-    tableInputEng2.playedGames = 3;
-    tableInputEng2.form = 'W,W,W';
-    tableInputEng2.won = 3;
-    tableInputEng2.draw = 0;
-    tableInputEng2.lost = 0;
-    tableInputEng2.points = 9;
-    tableInputEng2.goalsFor = 9;
-    tableInputEng2.goalsAgainst = 4;
-    tableInputEng2.goalDifference = 5;
-
-    // 3rd team
-    const tableInputEng3 = new TableInput();
-    tableInputEng3.position = 3;
-
-    const teamEng3 = new Team();
-    teamEng3.id = 62;
-    teamEng3.name = 'Everton FC';
-    teamEng3.crestUrl = 'https://crests.football-data.org/62.svg';
-
-    tableInputEng3.team = teamEng3;
-    tableInputEng3.playedGames = 3;
-    tableInputEng3.form = 'W,W,W';
-    tableInputEng3.won = 3;
-    tableInputEng3.draw = 0;
-    tableInputEng3.lost = 0;
-    tableInputEng3.points = 9;
-    tableInputEng3.goalsFor = 8;
-    tableInputEng3.goalsAgainst = 3;
-    tableInputEng3.goalDifference = 5;
-
-    const tableInputEng4 = new TableInput();
-    tableInputEng4.position = 4;
-
-    const teamEng4 = new Team();
-    teamEng4.id = 58;
-    teamEng4.name = 'Aston Villa FC';
-    teamEng4.crestUrl = 'https://crests.football-data.org/58.svg';
-
-    // 4th team
-    tableInputEng4.team = teamEng4;
-    tableInputEng4.playedGames = 2;
-    tableInputEng4.form = 'W,W';
-    tableInputEng4.won = 2;
-    tableInputEng4.draw = 0;
-    tableInputEng4.lost = 0;
-    tableInputEng4.points = 6;
-    tableInputEng4.goalsFor = 4;
-    tableInputEng4.goalsAgainst = 0;
-    tableInputEng4.goalDifference = 4;
-
-    // 5th team
-    const tableInputEng5 = new TableInput();
-    tableInputEng5.position = 5;
-
-    const teamEng5 = new Team();
-    teamEng5.id = 57;
-    teamEng5.name = 'Arsenal FC';
-    teamEng5.crestUrl = 'https://crests.football-data.org/57.svg';
-
-    tableInputEng5.team = teamEng5;
-    tableInputEng5.playedGames = 2;
-    tableInputEng5.form = 'W,W';
-    tableInputEng5.won = 2;
-    tableInputEng5.draw = 0;
-    tableInputEng5.lost = 0;
-    tableInputEng5.points = 6;
-    tableInputEng5.goalsFor = 4;
-    tableInputEng5.goalsAgainst = 0;
-    tableInputEng5.goalDifference = 4;
-
-    // standing
-    standingEng.table = new Array<TableInput>();
-    standingEng.table.push(tableInputEng1, tableInputEng2, tableInputEng3, tableInputEng4, tableInputEng5);
-
-    this.standingInput = new StandingInput();
-    this.standingInput.competition = competitionEng;
-    this.standingInput.season = seasonEng;
-    this.standingInput.standings = new Array<Standing>();
-    this.standingInput.standings.push(standingEng);
+  getLeagueTeams(competitionId: number): Observable<Array<Team>>{
+    return this.http.get<Array<Team>>('/api/teams/league-teams?league_id=' + competitionId)
+      .pipe(
+        map(data => {
+          const teamList = new Array<Team>();
+          for (const team of data){
+            teamList.push(Team.fromHttp(team));
+          }
+          return teamList;
+        })
+      );
```

Comment:
Ponownie - jeśli definiujesz coś jako Array<Team> w http.get to powinno to być dostępne od razu w odpowiedzi jako Array<Team> w subscription - ten pipe z mapą powinien być całkowicie zbędny.